### PR TITLE
Bug/4767 - Refactor destructuring useSelect results to fix the `Cannot read properties of undefined` JS error

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #4767 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by tofumatt — CHANGES_REQUESTED** [review]
Overall looks good! 👍🏻 

There's a few places where the changes resulted in the selects not working as previous… one was a missing `return` value which I fixed, but there's another one where changing a value to default to an object messed with a truthy check.

Also: let's change https://github.com/google/site-kit-wp/blob/2de75e88d8d077da2153617eb7afb1352e051960/assets/js/components/user-input/UserInputApp.js#L38-L41 while we're at it. It looks like it's calling `select( CORE_USER ).getUserInputSettings()` as a side effect, so we can probably just do:

```js
const hasFinishedGettingInputSettings = useSelect( ( select ) => {
	// This needs to be called here to check on its resolution,
	// as it's called/used by child components of this component,
	// but we need to call it here to know if it's resolving.
	//
	// This is sort of a select side-effect, but it's necessary here.
	select( CORE_USER ).getUserInputSettings();

	return select(
		CORE_USER
	).hasFinishedResolution( 'getUserInputSettings' );
} );
```

After that this should be good-to-go 🙂

File: assets/js/components/settings/SettingsRenderer.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default function SettingsRenderer( { slug } ) {
 		moduleLoaded,
 		connected,
 	} = useSelect( ( select ) => {
-		const module = select( CORE_MODULES ).getModule( slug );
+		const module = select( CORE_MODULES ).getModule( slug ) || {};
 		return {
 			...module,
 			moduleLoaded: !! module,
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/components/settings/SettingsRenderer.js:51]: That's 💯. My bad, I missed it. Thanks for catching it!

File: assets/js/components/ViewOnlyMenu/Service.js [hunk 2]
```diff
@@ -43,8 +43,8 @@ export default function Service( { module } ) {
 		select( CORE_USER ).hasCapability( PERMISSION_AUTHENTICATE )
 	);
 
-	const { name, owner } = useSelect( ( select ) =>
-		select( CORE_MODULES ).getModule( module )
+	const { name, owner } = useSelect(
+		( select ) => select( CORE_MODULES ).getModule( module ) || {}
 	);
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/components/ViewOnlyMenu/Service.js:48]: Yes, I was thinking of separating the calls. However, having the fallback seems cleaner as pointed out 👍

File: assets/js/modules/analytics/components/common/PropertySelect.js [hunk 3]
```diff
@@ -35,31 +35,28 @@ import useViewContext from '../../../../hooks/useViewContext';
...
+		return select( MODULES_ANALYTICS ).getProperties( accountID );
+	} );
 
-			return data;
+	const isResolvingProperties = useSelect( ( select ) => {
+		if ( hasModuleAccess === false || ! accountID ) {
+			return false;
 		}
-	);
+		select( MODULES_ANALYTICS ).isResolving( 'getProperties', [
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics/components/common/PropertySelect.js]: Good catch. Thanks for fixing it!

**Comment by hussain-t** [comment]
VRT is failing not due to the changes we made in the PR. However, I verified it's failing on `develop` as well.

**Review by tofumatt — APPROVED** [review]

**Pull request merged**